### PR TITLE
Pin Req version to 0.3.1

### PR DESCRIPTION
**Pull request by andrew-combs**
A change in Req (commit `a9df5fb6`) tracks which request steps have been run so they are not run a second time. This causes `get_partitions` to fail, because we rely on the previous `Request` struct when creating the actual request calls when fetching partitions.

This PR pins the req dependency to `0.3.1` until we can appropriately re-create a Request struct in the `get_partitions` step.

### CODE REVIEW

File: lib/avalanche/steps/get_partitions.ex [hunk 1]
```diff
@@ -75,10 +75,18 @@ defmodule Avalanche.Steps.GetPartitions do
 
   # reuse current request and turn it into a `StatusRequest`
   defp build_status_request(%Req.Request{} = request, path, partition, row_types) do
+    %{path: path} = URI.parse(path)
+
+    url =
+      request.url
+      |> URI.parse()
+      |> Map.put(:path, path)
+      |> Map.put(:query, "async=false&partition=#{partition}")
```
Comment (forest) [inline lib/avalanche/steps/get_partitions.ex:84]: I think you want to remove this. `partition` is added below and `async=false` is the default.

Also, I think `async` can be passed by the user as an option so we don't want to force it.

Comment (andrew-combs) [reply lib/avalanche/steps/get_partitions.ex:84]: Oops, good catch 👍🏼 thanks!

File: mix.exs [hunk 2]
```diff
@@ -47,7 +47,8 @@ defmodule Avalanche.MixProject do
       {:nimble_options, "~> 0.4.0"},
       {:mentat, "~> 0.7.1"},
       {:plug, "~> 1.13"},
-      {:req, "~> 0.3.0"},
+      # {:req, "~> 0.3.0"},
```
Comment (forest) [inline mix.exs:50]: I think you can do `{:req, "~> 0.3.1"}` too?

Comment (andrew-combs) [reply mix.exs:50]: That will automatically update it to `0.3.2`, but apparently this works: `{:req, "0.3.1"}`. That'll force the specific version.

I'll make that update 👍🏼

**Review by pdgonzalez872 — APPROVED** [review]
❤️

**Pull request merged**